src/algorithms/calorimetry/CalorimeterIslandCluster.cc

```diff
@@ -135,7 +135,9 @@ void CalorimeterIslandCluster::init() {
     };
 
     if (m_cfg.readout.empty()) {
-      error("readoutClass is not provided, it is needed to know the fields in readout ids");
+      if ((!m_cfg.adjacencyMatrix.empty()) || (!m_cfg.peakNeighbourhoodMatrix.empty())) {
+        throw std::runtime_error("readoutClass is not provided, it is needed to know the fields in readout ids");
```

Comment:
The config variable is not named `readoutClass`, so the error is confusing.
```suggestion
        throw std::runtime_error("'readout' is not provided, it is needed to know the fields in readout ids");
```